Cargo.toml

```diff
@@ -1,6 +1,6 @@
 [package]
 name = "qiskit-terra"
-version = "0.23.0"
+version = "0.24.0"
 edition = "2021"
```

Comment:
Do you want to trigger an update to `Cargo.lock` as well?

Reply:
Oh fair point, I skipped all of that because I was planning to do a full `cargo update` with #9393 but yeah we should bump the version number in the lock file. Let me do that right now.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/9396/commits/f72284d9aef1766555f83bc24e7af4692b2f4bf3